python/tk_multi_publish2/processing/plugin.py

```diff
@@ -129,6 +139,44 @@ def __init__(self, name, path, settings, logger):
 
         self._icon_pixmap = self._load_plugin_icon()
 
+    def _create_hook_instance(self, path):
+        """
+        Create the plugin's hook instance.
+
+        For publish plugins, we inject the plugin base hook's class in order to
+        provide a default implementation of the task settings UI.
+        """
+
+        plugin_base_file = "{self}/plugin_base.py"
+
+        # now try to create the hook instance using the
+        try:
+            # retrieve the class for the plugin base hook. this hook lives in
+            # the app itself, so we can refer to it with the hook expression
+            # that would be used if configured
+            plugin_base_class = self._bundle.get_hook_class(plugin_base_file)
```

Comment:
i think this is over complicated. I don't think we should think of the "base plugin" as a hook but as a normal class - probably not even keep it in the hooks folder - that way we can simply import it and we don't need this (imho a little weird!) method `bundle.get_hook_class`. The base plugin is a fixed interface and a class that is part of the publish code base and not a hook. Does that make sense?

Reply:
I originally wanted to do it this way, but my thought was that keeping it in the hook folder helped keep it visible to clients and made it discoverable.

Reply:
we should also think about sphinx docs - and hook docs in particular - just realised that by having a hook interface like this, and placing the class in the "normal" code, we can super easily sphinx document it!!! 😍

Reply:
Yeah. was thinking the same.